lib/src/extensions/list_extensions.dart

```diff
@@ -1,4 +1,68 @@
-extension CoreListExtensions<E> on List<E> {
+import 'package:collection/collection.dart';
+
+extension CoreIterableExtensions<T> on Iterable<T> {
+  /// Put [item] between elements of a non-empty iterable. Empty iterable is returned as is.
+  List<T> putBetween(T item) {
```

Comment:
Shouldn't it return `Iterable<T>`?

Reply:
yeah I thought so as well, but it anyhow is operating with a `List` inside. I thought it is reasonable to return a List, because callers would have to `toList()` again if they need a list.
returning a List spares this unnecessary `.toList()`